src/lib/Helpers.jsx

```diff
@@ -7,21 +6,32 @@ export const Spacers = styled.div`
   height: ${p => p.height ?? '0px'};
 `
 
-// Limits the rate at which a function can be called
-// Avoids too many renders on scroll for example
-export function debounce(func, wait, immediate) {
+// Returns a function, that, as long as it continues to be invoked, will not
+// be triggered. The function will be called after it stops being called for
+// `wait` milliseconds.
+export const debounce = (func, wait) => {
   let timeout
 
-  return function () {
-    const context = this
-    const args = arguments
-    const later = function () {
+  // This is the function that is returned and will be executed many times
+  // We spread (...args) to capture any number of parameters we want to pass
+  return function executedFunction(...args) {
```

Comment:
Can we change this to be `return (...args) => { ...`? Doesn't look like `executedFunction` is specifically referenced anywhere.